### ISSUE TITLE
Construct a SubGraph on an intermediate tensor

### ISSUE THREAD

**Issue by bubbles1990** [ARM-software/ComputeLibrary#376]
hi,

In the graph API: is it possible to construct a sub-graph on top of a graph intermediate layer?
I would like to branch out from some point in the main graph to another subgraph,
pseudo graph for illustration:
![image](https://user-images.githubusercontent.com/34748893/36895179-a61beb34-1e16-11e8-9011-67cfbc611479.png)

This maybe can be done by obtaining a pointer to the graph's intermediate tensor and use it as an input to some Node or Layer, but I can't figure out how to do that.

Thank you

**Comment by GeorgeARM** [comment]
Hello @bubbles1990,
Yes I think that this will work. You just have to provide an output tensor to the one subgraph and the branch should ignore it in the merge.
Haven't tested it thought, but its worth a try.

**Comment by bubbles1990** [comment]
Thank you @GeorgeARM for the quick reply,
I didn't completely understand what you mean though. Can you please elaborate or give a simple example?

Thanks in advance

**Comment by GeorgeARM** [comment]
Sure, something like this:
```
SubGraph i_a;
i_a << Layer1 << Layer2 << Tensor(Out0Accessor);

SubGraph i_ba;
i_ba << Layer5 << Tensor(Out1Accessor);

SubGraph i_bb;
i_bb << Layer6 << Tensor(Out2Accessor);

SubGraph i_b;
i_b << Layer3 << Layer4 << BranchLayer(i_ba, i_bb);

Gragh g;
g << Layer0 << BranchLayer(i_a, i_b) << Tensor(nullptr);
```

Or you can instead merge the branch at `Layer3`, meaning one branch will run the branch with the `Out0` tensor while the other just `Layer3`, so that you can early merge the branch and work on the main graph stream and not go into subgraph of subgraphs
For example:
```
SubGraph i_a;
i_a << Layer1 << Layer2 << Tensor(Out0Accessor);

SubGraph i_b;
i_b << Layer3 ;

SubGraph i_ba;
i_ba << Layer5 << Tensor(Out1Accessor);

SubGraph i_bb;
i_bb << Layer6 << Tensor(Out2Accessor);

Gragh g;
g << Layer0 << BranchLayer(i_a, i_b) << Layer4 << BranchLayer(i_ba, i_bb) << Tensor(nullptr);
```
My only concern is the last output tensor. In order for the graph to be finalized you need to provide an output, but you are inside a branch in your example. I added a dummy tensor in the end with no accessor. Not quite sure if this part will work though. Give it a try and if it fails I ll try create a test example and run it.

**Comment by bubbles1990** [comment]
Thank you very much for you comment @GeorgeARM ,

I've tried to run a test example as you suggested, but I ran into two issues:

1. It seems that BranchLayer accepts only _DEPTH_CONCATENATE_ merging method, while in my case I think _no merging_ is required.

2. After ending each branch with an output accessor, adding another dummy accessor in the end of the graph results in the error:
`ERROR in data_size_from_type ./arm_compute/core/Utils.h:115: Invalid data type No such file or directory`
